x/mint/abci_test.go

```diff
@@ -0,0 +1,175 @@
+package mint
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/burnt-labs/xion/x/mint/types"
+	minttypes "github.com/burnt-labs/xion/x/mint/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBeginBlocker(t *testing.T) {
+	type expected struct {
+		annualProvisions sdk.Dec
+		bondedRatio      sdk.Dec
+		burnedAmount     uint64
+		collectedAmount  uint64
+		inflation        sdk.Dec
+		minted           uint64
+		needed           uint64
+	}
+	type parameters struct {
+		bonded        sdk.Int
+		bondedRatio   sdk.Dec
+		fees          sdk.Coins
+		collectedFees sdk.Coin
+		burn          bool
+		mint          bool
+	}
+
+	stakingTokenSupply := sdk.NewIntFromUint64(100000000000)
+
+	tt := []struct {
+		name       string
+		parameters parameters
+		expected   expected
+	}{
+		{
+			name: "full bonded tokens",
+			parameters: parameters{
+				bonded:        stakingTokenSupply,
+				bondedRatio:   sdk.NewDecWithPrec(1, 4),
+				fees:          sdk.NewCoins(sdk.NewCoin("stake", sdk.NewInt(109))),
+				collectedFees: sdk.NewCoin("stake", sdk.NewInt(1000)),
+				mint:          true,
+				burn:          false,
+			},
+			expected: expected{
+				annualProvisions: sdk.NewDecWithPrec(7000000000, 0),
+				bondedRatio:      sdk.NewDecWithPrec(1, 4),
+				burnedAmount:     0,
+				collectedAmount:  1000,
+				inflation:        sdk.NewDecWithPrec(7, 2),
+				minted:           109,
+				needed:           1109,
+			},
+		},
+		{
+			name: "less than ideal bonded tokens",
+			parameters: parameters{
+				bonded:        sdk.NewInt(int64(100000000000 * 0.33)),
+				bondedRatio:   sdk.NewDecWithPrec(33, 2),
+				fees:          sdk.NewCoins(sdk.NewCoin("stake", sdk.NewInt(1045))),
+				collectedFees: sdk.NewCoin("stake", sdk.NewInt(0)),
+				mint:          true,
+				burn:          false,
+			},
+			expected: expected{
+				annualProvisions: sdk.NewDec(6600000000),
+				bondedRatio:      sdk.NewDecWithPrec(33, 2),
+				burnedAmount:     0,
+				collectedAmount:  0,
+				inflation:        sdk.NewDecWithPrec(20, 2),
+				minted:           1045,
+				needed:           1045,
+			},
+		},
+		{
+			name: "above staking threshold, feegrant has values",
```

Comment:
Feegrant is a different module for paying other peoples' fees, is it not? fee collector is what I think you mean

Reply:
That's correct. will fix.